### PR TITLE
Fixed Samba Move Object

### LINKED ISSUE #194: SAMBA AD: LDAP Result Code 53 "Unwilling To Perform": 00002035: Unwilling to perform. Old RDN must be deleted
Having issues with using ModifyDNRequest against a SAMBA active directory domain controller.  I believe its an encoding issue for value of TRUE.

There is a small deviation  in the LDAP specification from the BER encoding rules for a boolean. 

> The protocol elements of LDAP SHALL be encoded for exchange using the
>    Basic Encoding Rules [BER] of [ASN.1] with the following
>    restrictions:
> 
>    - Only the definite form of length encoding is used.
> 
>    - OCTET STRING values are encoded in the primitive form only.
> 
>    - If the value of a BOOLEAN type is true, the encoding of the value
>      octet is set to hex "FF".

Section 5.1 https://docs.ldap.com/specs/rfc4511.txt

Here a network dump of the failed request. 

screenshot_2018-11-07_13-40-10

### PR DESCRIPTION
**Pull request by ZDuygu**
This pr fixes #194

### CODE REVIEW

**Comment by mertcelen** [comment]
We need this pr to merged ASAP.

File: moddn.go [hunk 1]
```diff
@@ -39,7 +39,13 @@ func (req *ModifyDNRequest) appendTo(envelope *ber.Packet) error {
 	pkt := ber.Encode(ber.ClassApplication, ber.TypeConstructed, ApplicationModifyDNRequest, nil, "Modify DN Request")
 	pkt.AppendChild(ber.NewString(ber.ClassUniversal, ber.TypePrimitive, ber.TagOctetString, req.DN, "DN"))
 	pkt.AppendChild(ber.NewString(ber.ClassUniversal, ber.TypePrimitive, ber.TagOctetString, req.NewRDN, "New RDN"))
-	pkt.AppendChild(ber.NewBoolean(ber.ClassUniversal, ber.TypePrimitive, ber.TagBoolean, req.DeleteOldRDN, "Delete old RDN"))
+//	pkt.AppendChild(ber.NewBoolean(ber.ClassUniversal, ber.TypePrimitive, ber.TagBoolean, req.DeleteOldRDN, "Delete old RDN"))
```
Comment (stefanmcshane) [inline moddn.go:42]: Please remove the commented code from the PR

File: go.mod [hunk 2]
```diff
@@ -1,4 +1,4 @@
-module github.com/go-ldap/ldap
+module github.com/Zduygu/ldap
```
Comment (stefanmcshane) [inline go.mod]: Please revert this to the original name - I understand that this was for testing

File: v3/go.mod [hunk 3]
```diff
@@ -1,4 +1,4 @@
-module github.com/go-ldap/ldap/v3
+module github.com/ZDuygu/ldap/v3
```
Comment (stefanmcshane) [inline v3/go.mod:1]: Please revert this change - I understand it was for testing

File: moddn.go (hunk 1 above)
Comment (stefanmcshane) [reply moddn.go:42]: Please reflect this change in the v3/ folder also

File: go.mod [hunk 4]
```diff
@@ -1,4 +1,4 @@
-module github.com/go-ldap/ldap
+module github.com/ZDuygu/ldap
```
Comment (stefanmcshane) [inline go.mod:1]: Please revert this change - I understand it was for testing

**Review by stefanmcshane — CHANGES_REQUESTED** [review]
A few minor changes to be made before approval
Tag me and I will help get this approved for you asap

**Comment by mertcelen** [comment]
@stefanmcshane changes applied :)

**Comment by mertcelen** [comment]
@stefanmcshane you are absolutely right to ask for tests but if you check closer, there seems to be a bug with older go version tests itself. If I'm not wrong, it has nothing to do with our change. I'll gladly help if that's the case but since you flagged them as legacy and google obviously dropped support for it https://golang.org/doc/devel/release.html#policy we may as well just delete those.

Your call.

**Comment by stefanmcshane** [comment]
@mertcelen Whilst I wouldn't use Go's release history as an argument as the scopes are very different, I would agree that we should review the supported legacy versions going forward. This would need to be done as a separate PR, before merging this change.

@johnweldon Opinions on this? We would be potentially breaking backwards compatibility for any Go versions <= 1.6 (released 2016, support dropped 2017). 
However, I would suggest dropping support for 1.5 and 1.6 in a new PR, tagging as a hotfix release (v3.2.4), then merging this PR.

Aside: This would also provide a strong segue for tidying go modules up further, by dropping support for non-module installation, and making the new default branch "v3". If you agree to the above, I'll open a new issue for discussion on go module related updates.

**Review by stefanmcshane — APPROVED** [review]

**Comment by stefanmcshane** [comment]
@mertcelen @ZDuygu This has been tagged v3.2.4 and released. Thank you for your contributions and patience!

**Pull request merged**